dvc/command/config.py

```diff
@@ -85,6 +92,13 @@ def _get(self, remote, section, opt):
             [self.args.level] if self.args.level else self.config.LEVELS[::-1]
         )
 
+        if self.config.dvc_dir is None:
+            levels = [
+                level
+                for level in levels
+                if level not in self.config.REPO_ONLY_LEVELS
+            ]
```

Comment:
One other issue that I noticed - with the current PR, if the user explicitly calls `dvc config` with `--local` or `--project` from outside a DVC repo, the command succeeds and returns no output.

In this situation, I would expect the command to raise the "Not in a DVC repo" error instead of succeeding.

i.e. these commands should fail when run outside of a DVC repo:
```
dvc config --list --local
dvc config --list --project
dvc config --local core.analytics
dvc config --project core.analytics
```

For reference, the current master behavior for this gives the same unexpected error as `dvc config --list` with no explicit level:
```
$ dvc config --local core.analytics -v
2021-09-13 18:39:31,205 ERROR: unexpected error - 'local'
------------------------------------------------------------
Traceback (most recent call last):
  ...
  File "/Users/pmrowla/git/dvc/dvc/config.py", line 174, in _load_config
    filename = self.files[level]
KeyError: 'local'
------------------------------------------------------------
```

After this PR you get
```
dvc config --local core.analytics -v

```

But the lack of output here is not because `core.analytics` is unset - it happens because the entire
```py
for level in levels:
    conf = self.config.read(level)
    ...
```
loop is skipped (since `levels` is filtered out to be `levels = []` when `self.config.dvc_dir` is None).

The desired behavior would be to have these commands raise `ConfigError("Not inside a DVC repo")`.

Reply:
You are right. Throwing a config error would be an appropriate behavior. I'll add the changes. Also, I will add tests for the cases:
`dvc config --list --local`
`dvc config --list --project`
`dvc config --local core.analytics`
`dvc config --project core.analytics`